eval/main.py

```diff
@@ -219,47 +260,24 @@ def run(self, command) -> tuple[str, str, int]:
         print("--- Finished run ---\n")
         return stdout_full, stderr_full, p.returncode
 
-    def upload(self, files: Files):
-        for name, content in files.items():
-            path = self.working_dir / name
-            path.parent.mkdir(parents=True, exist_ok=True)
-            with open(path, "w") as f:
-                f.write(content)
-
-    def download(self) -> Files:
-        files = {}
-        ignore = [".git"]
-        for path in self.working_dir.glob("**/*"):
-            if any(path.match(i) for i in ignore):
-                continue
-            if path.is_file():
-                with open(path, "r") as f:
-                    try:
-                        content = f.read()
-                    except UnicodeDecodeError:
-                        content = "binary file"
-                    files[str(path.relative_to(self.working_dir))] = content
-        return files
-
 
 def execute(test: ExecTest) -> TestResult:
     """
     Executes the code.
     """
     print(f"Running test {test['name']} with prompt: {test['prompt']}")
-    env = GPTMeExecEnv()
+    agent = GPTMe()
```

Comment:
Reviewing myself, nitpick:

I realize we don't need to instantiate an object in this case.

The agent.act() can just be a function.